uber/site_sections/schedule.py

```diff
@@ -105,6 +106,40 @@ def schedule_tsv(self, session):
             'schedule': sorted(schedule.items(), key=lambda tup: c.ORDERED_EVENT_LOCS.index(tup[1][0]['location']))
         })
 
+    def ical(self, session, **params):
+        icalendar = ics.Calendar()
+
+        if 'locations' not in params or not params['locations']:
+            locations = [id for id, name in c.EVENT_LOCATION_OPTS]
+            calname = "all_locations"
+        else:
+            locations = json.loads(params['locations'])
+            calname = "_".join([name for id, name in c.EVENT_LOCATION_OPTS
+                                if str(id) in locations])\
+                .lower().replace(' ', '_')
+
+        for location in locations:
+            for event in session.query(Event)\
+                    .filter_by(location=int(location))\
+                    .order_by('start_time').all():
+                icalendar.events.add(ics.Event(
```

Comment:
Fun fact: the ics documentation says to use `calendar.events.append` to add events to a calendar.

The ics documentation is wrong.